packages/plugin-core/src/components/doctor/utils.ts

```diff
@@ -120,4 +124,37 @@ export class DoctorUtils {
       }
     }
   }
+
+  static async validateFilenameFromDocumentAndPromptIfNecessary(
+    document: vscode.TextDocument
+  ) {
+    const extension = ExtensionProvider.getExtension();
+    const wsUtils = extension.wsUtils;
+    const filename = path.basename(document.fileName, ".md");
+    const note = wsUtils.getNoteFromDocument(document);
+
+    if (!note) return true;
+
+    const isValid = NoteUtils.validateFname(filename);
+
+    if (isValid) return true;
+
+    const error = new DendronError({
+      message: `"${filename}" is not a valid hierarchy. Please rename the note so that every hierarchy is non-empty and doesn't contain leading or trailing whitespaces`,
+    });
+    VSCodeUtils.showMessage(
+      MessageSeverity.WARN,
+      error.message,
+      {},
+      { title: "Rename note" }
+    ).then(async (resp) => {
+      if (resp && resp.title === "Rename note") {
+        await wsUtils.openNote(note);
+        const cmd = new RenameNoteCommand(extension);
```

Comment:
does rename note work well with an invalid hierarchy? eg. will renaming `.foo` cause other errors?

Reply:
rename works as long as the renamed result is valid (which is what we want the users to do)

Reply:
I added a test for rename note for the invalid note case to be extra safe.